### DISCUSSION TITLE
Backing up "special" containers (ie with DBs etc)

### DISCUSSION THREAD

**Discussion by psyciknz** [djmaze/resticker#124] [General]
Trying to get my head around how people use various scripts for backing up restic.

Now I know you can, and are doing it myself, can back up my docker bind mount data (/hdd/docker-data) is where all my source docker-compose files and container configs live.  And I've moved all the docker volumes, overlays etc  form /var/lib/docker to /hdd/zdocker.

I'm backing both of those up with exclusions etc while everything is running.......but for all my DB based stuff, nextcloud, traccar, photoprisim, unifi etc, I get the impression that's the wrong way to do it.

So looking as to what i should do.
I have one container on my main docker host that backs up the aforementioned docker dirs.  
-Should (for best practice), I spin up new restic containers for nextcloud, photoprisim, unifi and include specific db backups and stop commands?  I'm conscious of the number of containers that will create?

Or I could via cron on the host system run a backup script that puts each db in the state it needs to be for backup - the downside to that, is that I like that this container package has it's own cron held in configuration files, so I dont have to look at cron on the system and the containers.   Ideally, schedule container, that can say, - hey at 0 30 3 0 * * * run this command, 0 30 4 * * * the main backup 

Thoughts and discussion?

**Comment by djmaze** [comment]
Yes, that is what I am doing as well, but using swarm mode and using swarm-cronjob as a scheduler. There is already some discussion on that here: https://github.com/djmaze/resticker/issues/115#issuecomment-1035508867

**Comment by psyciknz** [comment]
So you run your postgres backup container independently of your restic backup to output a DB dump where the backup can pick it up?

**Reply by djmaze** [reply]
Yes.

**Reply by psyciknz** [reply]
And do you leave that container running consuming some resources or shut it down and start it when needed?

That's my ideal.

**Reply by djmaze** [reply]
The container is started by swarm-cronjob. It does the dump and then exits.

(Btw, I will probably create a swarm-based version of resticker at some point which will be started by swarm-cronjob instead of having cron integrated.)

**Reply by psyciknz** [reply]
Ok.   I wasn't sure how that swarm for job fitted in...I guess I just need to see if that can run non Swarm mode

**Reply by djmaze** [reply]
A bit off-topic but I warmly recommend using swarm mode even for people who do not use the multi-host features. Very easy to setup and it's an improvement over compose-based deployment IMO. Just sayin..

**Reply by psyciknz** [reply]
Hmm ok.  I might have to look into that again.  I had tried swarm a while ago, but got hung up on trying to get persistent storage between nodes (things like portainer storage, caddy with certs).   Trying things like glusterfs.   It got a bit hard. 

Couple with using my dns on my router to point to a single address seemed a funny way of access the load balancing function of swarm.
